### PR TITLE
Optimization: sanitize html to avoid XSS attack #127

### PR DESCRIPTION
**Pull request by Jocs**
| Q                | A
| ---------------- | ---
| Bug fix?         | yes
| New feature?     | no
| BC breaks?       | no
| Deprecations?    | no
| New tests added? | not needed
| Fixed tickets    | #127 
| License          | MIT

### Description

[Description of the bug or feature]

Optimization: sanitize html to avoid XSS attack

--

### CODE REVIEW

File: src/editor/config.js [hunk 1]
```diff
@@ -180,6 +180,17 @@ export const TABLE_TOOLS = [{
   icon: 'icon-del'
 }]
 
+export const DOMPURIFY_CONFIG = {
+  FORBID_ATTR: ['style', 'class', 'contenteditable'],
```
Comment (Jocs) [inline src/editor/config.js]: @fxha Do you have any other attributes to forbid?

File: src/editor/contentState/codeBlockCtrl.js [hunk 2]
```diff
@@ -142,7 +145,8 @@ const codeBlockCtrl = ContentState => {
         if (block.functionType === 'html') {
           const preBlock = this.getNextSibling(block)
           const htmlBlock = this.getParent(this.getParent(block))
-          const escapedHtml = escapeInBlockHtml(block.text)
+          const escapedHtml = DOMPurify.sanitize(escapeInBlockHtml(block.text), DOMPURIFY_CONFIG)
+          console.log(escapedHtml)
```
Comment (fxha) [inline src/editor/contentState/codeBlockCtrl.js]: please remove this

Comment (Jocs) [reply src/editor/contentState/codeBlockCtrl.js]: ok

**Comment by fxha** [comment]
This doesn't work for me, but I also get no `console` output nor errors. 

**Steps to reproduce:** c&p one source text from #127 and insert it into source code mode.

**Comment by fxha** [comment]
> What's the meaning?

If I write `console.log` right after ` DOMPurify.sanitize(...)` I got no output. Which means that the function is never called.

Do you sanitize HTML code blocks only? You have to protect the whole markdown text file. E.g.:

```
foo

<script foo>alert('bar')</script> <--- this is loaded into editor

bar
```

**Comment by Jocs** [comment]
@fxha I did not sanitize `script`, `style`, `title`. in other words I don't make them disappeared.

I just escape the `script`, `style`, `title` tags as the Github do.

**Comment by Jocs** [comment]
@fxha I get you point 👍

**Comment by Jocs** [comment]
@fxha My fault, I'll update this PR.

**Comment by Jocs** [comment]
@fxha PR updated.

**Pull request merged**